core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -924,12 +923,12 @@ private[spark] object JsonProtocolSuite extends Assertions {
         assert(e1.jobId === e2.jobId)
         assertEquals(e1.jobResult, e2.jobResult)
       case (e1: SparkListenerEnvironmentUpdate, e2: SparkListenerEnvironmentUpdate) =>
-        assertEquals(e1.environmentDetails, e2.environmentDetails)
+        assertEquals(e1.environmentDetails.toMap, e2.environmentDetails.toMap)
       case (e1: SparkListenerExecutorAdded, e2: SparkListenerExecutorAdded) =>
-        assert(e1.executorId === e1.executorId)
+        assert(e1.executorId === e2.executorId)
         assertEquals(e1.executorInfo, e2.executorInfo)
       case (e1: SparkListenerExecutorRemoved, e2: SparkListenerExecutorRemoved) =>
-        assert(e1.executorId === e1.executorId)
+        assert(e1.executorId === e2.executorId)
```

Comment:
According to the PR description, after fixing this, `toMap` causes failure, right?

Reply:
No, two things have been done in this pr : one is code cleanup, the other is bug fix.

Is it better to divide this pr into two parts?

Sorry, my descriptive ability is poor. It seems that there is a misunderstanding ...

Reply:
> According to the PR description, after fixing this, `toMap` causes failure, right?

After fixing this, `toMap` is redundant collection conversion